### PR TITLE
feat(button): Add href to button to behave as a link

### PR DESCRIPTION
**Pull request by JohannesAnd**
Ref: https://app.asana.com/0/1200489836971088/1201799627233580/f

Add the option to give a href to the button component and make it work as a link

### CODE REVIEW

File: packages/component-library/src/components/plmg-button/plmg-button.scss [hunk 1]
```diff
@@ -278,6 +278,10 @@
   font-style: normal;
   font-weight: tokens.$plmg-font-weight-regular;
 
+  // Remove link styling
+  text-decoration: none;
+  width: fit-content;
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-button/plmg-button.scss:283]: Why do you need to set the width to fit-content?

File: packages/component-library/src/components/plmg-button/plmg-button.tsx [hunk 2]
```diff
@@ -128,19 +128,57 @@ export class Button {
...
+      throw new Error('href must be a string');
+  }
+  /**
+   * Define links rel
+   */
+  @Prop() rel: string;
+  @Watch('rel')
+  validateRel(newValue: string) {
+    if (newValue && !this.href)
+      throw new Error('button must have a href to have a rel');
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-button/plmg-button.tsx:147]: :+1:

File: packages/component-library/src/components/plmg-button/plmg-button.scss (hunk 1 above)
Comment (JohannesAnd) [reply packages/component-library/src/components/plmg-button/plmg-button.scss:283]: Because it made the a-tag full width without

Comment (TanguyDucky) [reply packages/component-library/src/components/plmg-button/plmg-button.scss:283]: Just realized it's not compatible with IE :confused: 
We did not really decide what was our position regarding supporting older browsers.

**Pull request merged**